src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -32,6 +33,91 @@ internal static partial class PathInternal
         };
 
         /// <summary>
+        /// Adds the extended path prefix (\\?\) if not already present.
+        /// </summary>
+        internal static string AddExtendedPathPrefix(string path)
+        {
+            if (IsExtended(path))
+                return path;
+
+            if (IsExtendedUnc(path))
+                return path.Insert(2, PathInternal.UncExtendedPrefixToInsert); // Given \\server\share in longpath becomes \\?\UNC\server\share  => UNCLongPathPrefix + path.SubString(2); => The actual command simply reduces the operation cost.
```

Comment:
I can create a new test project in \Common\Tests that actually includes these shared files and better unit test these helpers without needing InternalsVisibleTo. I've been mulling it over for a few days and I think it makes a lot of sense.